packages/medusa/src/services/order-edit.ts

```diff
@@ -0,0 +1,127 @@
+import { EntityManager } from "typeorm"
+import { FindConfig } from "../types/common"
+import { buildQuery } from "../utils"
+import { MedusaError } from "medusa-core-utils"
+import { OrderEditRepository } from "../repositories/order-edit"
+import {
+  LineItem,
+  OrderEdit,
+  OrderEditItemChangeType,
+  OrderItemChange,
+} from "../models"
+import { TransactionBaseService } from "../interfaces"
+import { OrderService } from "./index"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  orderEditRepository: typeof OrderEditRepository
+  orderService: OrderService
+}
+
+export default class OrderEditService extends TransactionBaseService {
+  protected transactionManager_: EntityManager | undefined
+  protected readonly manager_: EntityManager
+  protected readonly orderEditRepository_: typeof OrderEditRepository
+  protected readonly orderService_: OrderService
+
+  constructor({
+    manager,
+    orderEditRepository,
+    orderService,
+  }: InjectedDependencies) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.orderEditRepository_ = orderEditRepository
+    this.orderService_ = orderService
+  }
+
+  async retrieve(
+    orderEditId: string,
+    config: FindConfig<OrderEdit> = {}
+  ): Promise<OrderEdit | never> {
+    const orderEditRepository = this.manager_.getCustomRepository(
+      this.orderEditRepository_
+    )
+    const { relations, ...query } = buildQuery({ id: orderEditId }, config)
+
+    const orderEdit = await orderEditRepository.findOneWithRelations(
+      relations as (keyof OrderEdit)[],
+      query
+    )
+
+    if (!orderEdit) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `Order edit with id ${orderEditId} was not found`
+      )
+    }
+
+    return orderEdit
+  }
+
+  async computeLineItems(orderEdit: OrderEdit): Promise<OrderEdit> {
```

Comment:
thought(non-blocking): don't have strong opinions on this but would like to hear what you think:

does it make sense to change this to take the id of the OrderEdit instead of relying on the data to be correctly formatted in the param? 

This would allow us to do something like this:
1. Retrieve minimal OrderEdit details: `this.retrieve(id, { select: ["id", "order_id"] })`
2. Retrieve all OrderEdit changes e.g. `changeService.list({ order_edit_id: id })`
3. Retrieve order items `lineItemService.list({ order_id: orderEdit.order_id, fulfilled_quantity: 0 })`
4. The above should suffice for computing the items
5. The method could simply return the computed items

My thought comes arises from trying to avoid the slow mess built up in the TotalsService where if we don't join and pass correct data to the getTotals,getShippingTotal,etc. methods we get inconsistent results;

Reply:
I like that, maybe a bit revisited 😀 Why is there a condition on fulfilled_quantity to 0?

Reply:
Because you can't edit LineItems that have already been fulfilled so this would be a way to eliminate those - maybe not the best approach (as partially fulfilled items would still make sense to show) just a rough suggestion

Reply:
Ignore the fulfilled condition for now; think it makes sense to show all items and then somewhere else enforce that fulfilled items can't be removed or quantity reduced below the fulfilled_quantity

Reply:
yeah I think that this check will be added as part of the line item methods manipulation wdyt?